matching-service/src/app.ts

```diff
@@ -0,0 +1,43 @@
+import express from "express";
+import cors from "cors";
+import * as dotenv from "dotenv";
+
+import messageRoutes from "./routes/messageRoute";
+import { initRabbitMQ } from "./services/rabbitMqService";
+import redisClient from "./config/redisConfig";
+import { startBackgroundTransfer } from "./services/matchingService";
+
+dotenv.config({ path: ".env.dev" });
+
+const app = express();
+const PORT = process.env.PORT || 5001; // 5001 to prevent conflicts
+const QUEUE = process.env.MATCHING_SERVICE_QUEUE || "matching-service";
+
+app.use(express.json());
+app.use(cors()); // config cors so that front-end can use
+app.options("*", cors());
+
+const apiVersion = "/api/v1";
+
+// Allow frontend to talk to service
+app.use(cors({ origin: true, credentials: true }));
+
+app.use(`${apiVersion}/message`, messageRoutes);
+
+// ping server
+app.get(`${apiVersion}/ping`, (req, res) => {
```

Comment:
I think we should remove all routes to this matching_service. A listening service shouldnt require anyone to tell him to get the mailbox (at least in gamedev). Rabbitmq's consume function automatically prevents the program from executing further.

Instead, there should be a while(True) loop that constantly does the following:

1. Consume/wait for incoming messages in the RabbitMQ queue
2. Check for if the new object has any match, if no match, then put that object in a holding area/local queue
3. Send out any Redis kv pairs via RabbitMQ queue if matches are found.